src/Core/Module/Exception/ModuleErrorInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Module\Exception;
+
+/**
+ * If an exception that implements this ModuleErrorInterface
+ * is thrown, its message will be displayed to the end-user
+ */
+interface ModuleErrorInterface
```

Comment:
Ideally it should extend [Throwable](https://www.php.net/manual/en/class.throwable.php), then it wouldn't be empty. But it's not possible. It's just a marker interface so that you have more flexibility to create your own exceptions in this specific case.

From a DX standpoint, it's easier to use the exception message. Regarding `FlashErrorInterface` it could be integrated in the core later and be made a parent to this one.

Reply:
Ideally it should extend [Throwable](https://www.php.net/manual/en/class.throwable.php), then it wouldn't be empty. But it's not possible. It's just a marker interface so that you have more flexibility to create your own exceptions in this specific case.

From a DX standpoint, it's easier to use the exception message. Regarding `FlashErrorInterface` it could be integrated in the core later and be made a parent to this one.

Reply:
Ok let's do that for now then :)